server/src/main/java/io/pravega/schemaregistry/storage/client/HostStore.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.storage.client;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.control.impl.ControllerImpl;
+import io.pravega.client.control.impl.ControllerImplConfig;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.stream.api.grpc.v1.Controller;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * This is a temporary implementation to directly query controller for table segment's location. 
+ * Whenever table client is released, replace this class with table client.  
+ */
+class HostStore {
+    @Getter(AccessLevel.PACKAGE)
+    private final ControllerImpl controller;
+    private final Cache<String, Controller.NodeUri> cache;
+
+    HostStore(ClientConfig clientConfig, ScheduledExecutorService executor) {
+        controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig).build(), executor);
+        this.cache = CacheBuilder.newBuilder()
+                    .maximumSize(100)
+                    .build();
+    }
+
+    CompletableFuture<Controller.NodeUri> getHostForTableSegment(String tableName) {
+        try {
+            Controller.NodeUri uri = cache.getIfPresent(tableName);
+            if (uri == null) {
+                return controller.getEndpointForSegment(tableName)
+                                 .thenApply(u -> {
+                                     Controller.NodeUri nodeUri = Controller.NodeUri.newBuilder().setEndpoint(u.getEndpoint()).setPort(u.getPort()).build();
+                                     cache.put(tableName, nodeUri);
+                                     return nodeUri;
+                                 })
+                        .exceptionally(e -> {
+                            throw StoreExceptions.create(StoreExceptions.Type.CONNECTION_ERROR, "Failed to contact controller");
+                        });
+            } else {
+                return CompletableFuture.completedFuture(uri);
+            }
+        } catch (Exception e) {
```

Comment:
actually we dont need this catch block. controller client returns a future which can fail. 
we have added this safety catch block in case controller client throws an exception in trying to submit the request and before creating and returning the future. in which case we do a catch all so that this method's caller is always returned a future. 
we just convert this to StoreException.Connection so that it can be retried.